### ISSUE TITLE
linter: oxlint -f github format missing file paths in annotations

### ISSUE THREAD

**Issue by nwparker** [oxc-project/oxc#17784]
## Summary

When using `oxlint -f github`, the output annotations are missing file path information, causing them to appear only in CI logs rather than as inline annotations on GitHub PR files.

## Current Behavior

Running oxlint with `-f github` format produces output like:

```
##[warning]File has too many lines (309).
```

This format lacks the `file=path,line=N` parameters required for GitHub to create inline PR annotations.

## Expected Behavior

The output should include file path and line information:

```
::warning file=src/components/MyComponent.tsx,line=309,col=1,title=eslint(max-lines)::File has too many lines (309).
```

This format allows GitHub Actions to create inline annotations on the specific files in a PR's "Files changed" tab.

## Comparison with `-f unix`

The `-f unix` format correctly includes file information:

```
src/components/MyComponent.tsx:309:1: File has too many lines (309). [Warning/eslint(max-lines)]
```

## Workaround

We created a shell script to convert `-f unix` output to proper GitHub annotation format:

```bash
#!/usr/bin/env bash
# Usage: oxlint -f unix [files...] 2>&1 | ./oxlint-to-github-annotations.sh [path-prefix]

PATH_PREFIX="${1:-}"

while IFS= read -r line; do
  if [[ $line =~ ^([^:]+):([0-9]+):([0-9]+):\ (.+)\ \[(Warning|Error)/([^\]]+)\]$ ]]; then
    file="${PATH_PREFIX}${BASH_REMATCH[1]}"
    line_num="${BASH_REMATCH[2]}"
    col="${BASH_REMATCH[3]}"
    message="${BASH_REMATCH[4]}"
    severity="${BASH_REMATCH[5]}"
    rule="${BASH_REMATCH[6]}"

    if [[ $severity == "Error" ]]; then
      echo "::error file=${file},line=${line_num},col=${col},title=${rule}::${message}"
    else
      echo "::warning file=${file},line=${line_num},col=${col},title=${rule}::${message}"
    fi
  else
    echo "$line"
  fi
done
```

## Proposed Fix

Update the `-f github` formatter to include file path, line, and column information in the annotation output, following [GitHub's workflow command format](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-a-warning-message):

```
::warning file={path},line={line},col={col},title={rule}::{message}
::error file={path},line={line},col={col},title={rule}::{message}
```

## Environment

- oxlint version: 1.33.0 / 1.36.0
- OS: Ubuntu (GitHub Actions runner)

**Comment by nrayburn-tech** [comment]
The output looks correct to me when running locally. Is this something you're seeing locally, or do you only see the problem in GitHub actions?

```text
nrayburn@nrayburn-macbook nested-configs % npx oxlint@1.33.0 -f github .
::error file=index.js,line=6,endLine=6,col=10,endColumn=16,title=eslint(curly)::Expected { after 'if' condition.
::warning file=index.js,line=3,endLine=3,col=1,endColumn=9,title=eslint(no-debugger)::`debugger` statement is not allowed
::warning file=index.js,line=4,endLine=4,col=1,endColumn=14,title=eslint-plugin-unicorn(no-new-array)::Do not use `new Array(singleArgument)`.
::error file=package-a/index.js,line=2,endLine=2,col=1,endColumn=10,title=eslint(no-debugger)::`debugger` statement is not allowed
::error file=package-a/index.js,line=6,endLine=6,col=1,endColumn=9,title=eslint(no-debugger)::`debugger` statement is not allowed
```

**Comment by nrayburn-tech** [comment]
Example GitHub actions run here https://github.com/oxc-project/oxc/actions/runs/20887204371/job/60012296112?pr=17859. As far as I can tell it is working correctly there as well.

**Comment by nwparker** [comment]
I can't seem to reproduce so I will close this. Thanks for look at it!
